thumbnail-service/src/main/java/no/sikt/nva/handler/ThumbnailRequestHandler.java

```diff
@@ -0,0 +1,144 @@
+import no.sikt.nva.thumbnail.ThumbnailerException;
+import no.sikt.nva.thumbnail.ThumbnailerManager;
+import no.unit.nva.s3.S3Driver;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.StringUtils;
+import nva.commons.core.attempt.Failure;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+
+public class ThumbnailRequestHandler implements RequestHandler<S3Event, Void> {
+
+    public static final String COULD_NOT_CREATE_THUMBNAIL_LOG_MESSAGE = "Could not create thumbnail";
+    public static final String CURRENTLY_SUPPORTING_THE_FOLLOWING_MIME_TYPES_LOG_MESSAGE = "Currently Supporting the "
+                                                                                           + "following Mime Types: ";
+    public static final String FILENAME_PREFIX = "filename=\"";
+    public static final String FILENAME_POSTFIX = "\"";
+    private static final Logger logger = LoggerFactory.getLogger(ThumbnailRequestHandler.class);
+
+    // protects against overwriting existing files in
+    // directory, plus it's the only place aws allows filewriting; To view thumbnails created locally: replace with
+    // "" and comment out outputFile.deleteOnExit() / inputFile.deleteOnExit();
+    private static final String INPUT_FILE_NAME_PREFIX = "/tmp/";
+    private static final String OUTPUT_FILE_NAME = "thumbnail.png";
+    private static final String THUMBNAIL_BUCKET_ENVIRONMENT_FIELD = "THUMBNAIL_BUCKET";
+    public static final String DELIMITER = ", ";
+    private final String thumbnailBucketName;
+    private final S3Client s3Client;
+    private File inputFile;
+    private String mimeTypeFromS3Response;
+
+    @JacocoGenerated
+    public ThumbnailRequestHandler() {
+        this(S3Driver.defaultS3Client().build());
+    }
+
+    public ThumbnailRequestHandler(S3Client s3Client) {
+        this.thumbnailBucketName = new Environment().readEnv(THUMBNAIL_BUCKET_ENVIRONMENT_FIELD);
+        this.s3Client = s3Client;
+    }
+
+    @Override
+    public Void handleRequest(S3Event s3Event, Context context) {
+        inputFile = readFile(s3Event);
+        var outputFile = generateThumbnail();
+        writeThumbnailToS3(outputFile);
+        outputFile.deleteOnExit();
+        inputFile.deleteOnExit();
+        return null;
+    }
+
+    private void writeThumbnailToS3(File outputFile) {
+        try {
+            s3Client.putObject(PutObjectRequest.builder()
+                                   .bucket(thumbnailBucketName)
+                                   .key("keyForFeature(featureState.getFeature())")
```

Comment:
the string literal here needs some explanation because it seems to be a string that later will become a method call in some mysterious way.

Reply:
Nice catch, that needs to be replaced with the input-files object key. Right now the thumbnailer handler sits an listen to s3 events, it does not know anything about the publication-metadata so it cannot update a post in the database. So best it can do is store the thumbnail with the same key as the input-file but in different bucket.

Reply:
This can be changed later on I suppose, when we know better how we would like to use it. Ideally this should be triggered from a backend service that can call the thumbnail handler, and update the metadata about a post that there is a thumbnail file with key blahblah. 

To do that we need to solve the following then: the backend needs to know when the frontend is done uploading to s3. I am guessing a task later on.